transport_analysis/analysis/velocityautocorr.py

```diff
@@ -5,111 +5,190 @@
-    results: :class:`~MDAnalysis.analysis.base.Results`
-        results of calculation are stored here, after calling
-        :meth:`VelocityAutocorr.run`
-    start: Optional[int]
-        The first frame of the trajectory used to compute the analysis
-    stop: Optional[int]
-        The frame to stop at for the analysis
-    step: Optional[int]
-        Number of frames to skip between each analyzed frame
-    n_frames: int
-        Number of frames analysed in the trajectory
-    times: numpy.ndarray
-        array of Timestep times. Only exists after calling
-        :meth:`VelocityAutocorr.run`
-    frames: numpy.ndarray
-        array of Timestep frame indices. Only exists after calling
-        :meth:`VelocityAutocorr.run`
+    dim_fac : int
+        Dimensionality :math:`d` of the VACF.
+    results.timeseries : :class:`numpy.ndarray`
+        The averaged VACF over all the particles with respect to lag-time.
+        Obtained after calling :meth:`VelocityAutocorr.run`
+    results.vacf_by_particle : :class:`numpy.ndarray`
+        The VACF of each individual particle with respect to lag-time.
+    start : Optional[int]
+        The first frame of the trajectory used to compute the analysis.
+    stop : Optional[int]
+        The frame to stop at for the analysis.
+    step : Optional[int]
+        Number of frames to skip between each analyzed frame.
+    n_frames : int
+        Number of frames analysed in the trajectory.
+    n_particles : int
+        Number of particles VACF was calculated over.
     """
 
     def __init__(
         self,
-        universe_or_atomgroup: Union["Universe", "AtomGroup"],
-        select: str = "all",
-        # TODO: add your own parameters here
+        atomgroup: "AtomGroup",
+        dim_type: str = "xyz",
+        fft: bool = True,
         **kwargs
-    ):
+    ) -> None:
         # the below line must be kept to initialize the AnalysisBase class!
-        super().__init__(universe_or_atomgroup.trajectory, **kwargs)
+        super().__init__(atomgroup.universe.trajectory, **kwargs)
         # after this you will be able to access `self.results`
         # `self.results` is a dictionary-like object
         # that can should used to store and retrieve results
         # See more at the MDAnalysis documentation:
         # https://docs.mdanalysis.org/stable/documentation_pages/analysis/base.html?highlight=results#MDAnalysis.analysis.base.Results
 
-        self.universe = universe_or_atomgroup.universe
-        self.atomgroup = universe_or_atomgroup.select_atoms(select)
+        if isinstance(atomgroup, UpdatingAtomGroup):
+            raise TypeError("UpdatingAtomGroups are not valid for VACF "
+                            "computation")
+
+        # args
+        self.dim_type = dim_type
+        self._parse_dim_type()
+        self.fft = fft
+
+        # local
+        self.atomgroup = atomgroup
+        self.n_particles = len(self.atomgroup)
+        self._velocity_array = None
```

Comment:
Agree with @orionarcher, lets just allocate the arrays here rather than in _prepare.

Reply:
So I tried this and it seems like it does not work because `n_frames` is not accessible in `__init__()`

Reply:
If you're very keen on having empty arrays there, one option is to set up `np.empty((0, ...), dtype=float)`. That could be confusing for users though.

Reply:
If that is the only alternative, I feel the current setup would be a little nicer. @hmacdope @orionarcher What are your thoughts?

Reply:
Current setup meaning allocating them in _prepare instead of in __init__? I'd be happy with that option too. I like what you have here once the redundant `None` assignments are removed.

Reply:
Ah, you're right! They are redundant!